### PR TITLE
Add additional checks to NativeECDHKeyAgreement to match upstream

### PR DESCRIPTION
**Pull request by KostasTsiounis**
There are additional checks that are required during the `init` and `doPhase` parts of `ECDH` key agreement. Those checks are already present in the original `ECDHKeyAgreement` class, so they had to be added to the `NativeECDHKeyAgreement` class as well.

Back-ported from: https://github.com/ibmruntimes/openj9-openjdk-jdk/pull/788

Signed-off by: Kostas Tsiounis [kostas.tsiounis@ibm.com](mailto:kostas.tsiounis@ibm.com)

### CODE REVIEW

**Comment by KostasTsiounis** [comment]
FYI @keithc-ca

File: closed/adds/jdk/src/share/classes/sun/security/ec/NativeECDHKeyAgreement.java [hunk 1]
```diff
@@ -25,27 +25,25 @@
...
-import java.security.PublicKey;
 import java.security.SecureRandom;
 import java.security.interfaces.ECKey;
+import java.security.interfaces.ECPublicKey;
 import java.security.spec.AlgorithmParameterSpec;
-import java.security.spec.ECGenParameterSpec;
 import java.security.spec.ECParameterSpec;
-import java.security.spec.InvalidParameterSpecException;
 import java.util.Map;
+import java.util.Optional;
```
Comment (keithc-ca) [inline closed/adds/jdk/src/share/classes/sun/security/ec/NativeECDHKeyAgreement.java:46]: Please remove this unused import.

File: closed/adds/jdk/src/share/classes/sun/security/ec/NativeECDHKeyAgreement.java [hunk 2]
```diff
@@ -71,6 +67,9 @@ public final class NativeECDHKeyAgreement extends KeyAgreementSpi {
     /* private key, if initialized */
     private ECPrivateKeyImpl privateKey;
 
+    /* operations associated with private key, if initialized */
+    private ECOperations privateKeyOps;
+
```
Comment (keithc-ca) [inline closed/adds/jdk/src/share/classes/sun/security/ec/NativeECDHKeyAgreement.java:72]: Please remove this unused field.

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: closed/adds/jdk/src/share/classes/sun/security/ec/NativeECDHKeyAgreement.java (hunk 1 above)
Comment (KostasTsiounis) [reply closed/adds/jdk/src/share/classes/sun/security/ec/NativeECDHKeyAgreement.java]: Removed.

File: closed/adds/jdk/src/share/classes/sun/security/ec/NativeECDHKeyAgreement.java (hunk 2 above)
Comment (KostasTsiounis) [reply closed/adds/jdk/src/share/classes/sun/security/ec/NativeECDHKeyAgreement.java]: Removed.

**Comment by keithc-ca** [comment]
Jenkins test sanity alinux jdk8

**Comment by KostasTsiounis** [comment]
The `TestEC` one is a known issue (see https://github.com/eclipse-openj9/openj9/issues/19479). I have the fix for that and you should see a PR soon.

The other one doesn't seem `ECDH` related. Maybe we should open another issue for that.

**Comment by keithc-ca** [comment]
Grinder for jdk_security3 passed in https://openj9-jenkins.osuosl.org/job/Grinder/3628 (running on the same machine as used by https://openj9-jenkins.osuosl.org/job/Grinder/3627).

**Review by keithc-ca — APPROVED** [review]

**Pull request merged**